ReactWindows/ReactNative/Modules/Launch/LauncherModule.cs

```diff
@@ -176,7 +205,36 @@ public void OnActivated(string url)
         /// <param name="url">The URL.</param>
         public static void SetActivatedUrl(string url)
         {
-            s_activatedUrl = url;
+            // Static state doesn't work well in RNW modules for scenarios that involve creating multiple instances of a
+            // module during the lifetime of the application. Examples:
+            //   1. An app creates multiple RNW bridges in which case each module will have multiple instances living at
+            //      the same time.
+            //   2. An app recreates the RNW context in which case each module will have multiple instances but
+            //      their lifetimes won't overlap.
+            // 
+            // This method is designed to work for (2). The idea is that `url` is intended for 1 module instance.
+            // If the instance is already around, the activated URL will be given to it. Otherwise, the module
+            // instance will *consume* the activated URL during initialization. Once consumed, the URL won't
+            // be around for other instances to consume during their initialization.
+            //
+            // The right behavior for (1) isn't clear.
```

Comment:
A potential solution for (1) is a mechanism for registering actions to take on modules after the instance is created.